src/main/java/com/thirdcc/webapp/service/impl/UserUniInfoServiceImpl.java

```diff
@@ -122,4 +122,22 @@ public boolean  isUserUniInfoCompleted(Long userId) {
 
         return hasCourseProgramId && hasYearSession && hasIntakeSemester && hasStayIn;
     }
+
+    @Override
+    public Optional<UserUniInfoDTO> getUserUniInfoByUserId(Long userId) {
+        return userUniInfoRepository
+            .findOneByUserId(userId)
+            .map(userUniInfoMapper::toDto);
+    }
+
+    @Override
+    public Optional<UserUniInfoDTO> mapUserUniInfoWithUser(UserUniInfoDTO userUniInfoDTO, User user) {
+        userUniInfoDTO.setFirstName(user.getFirstName());
+        userUniInfoDTO.setLastName(user.getLastName());
+        userUniInfoDTO.setGender(user.getGender());
+        userUniInfoDTO.setPhoneNumber(user.getPhoneNumber());
+        userUniInfoDTO.setDateOfBirth(user.getDateOfBirth());
+        userUniInfoDTO.setImageUrl(user.getImageUrl());
+        return Optional.of(userUniInfoDTO);
```

Comment:
why return optional if the userUniInfoDTO must be non null?

Reply:
resolved in 58a46524c85431fba01f968c73f49a401a5a6ad6